library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -207,6 +232,229 @@ libspdm_return_t libspdm_receive_response(void *context, const uint32_t *session
+
+        send_info->chunk_bytes_transferred += copy_size;
+        if (send_info->chunk_bytes_transferred >= send_info->large_message_size) {
+            spdm_request->header.param1 |= SPDM_CHUNK_SEND_REQUEST_ATTRIBUTE_LAST_CHUNK;
+        }
+
+        spdm_request_size = (chunk_ptr + copy_size) - (uint8_t*)spdm_request;
+        status = libspdm_send_request(
+            spdm_context, session_id, false,
+            spdm_request_size, spdm_request);
+
+        spdm_request = NULL;
+        spdm_request_size = 0;
+
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            break;
+        }
+
+        response = message;
+        response_size = message_size;
+
+        libspdm_zero_mem(response, response_size);
+
+        status = libspdm_receive_response(
+            spdm_context, session_id, false,
+            &response_size, &response);
+
+        if (LIBSPDM_STATUS_IS_ERROR(status)) {
+            break;
+        }
+        spdm_response = (void*) (message);
+
+        if (response_size < sizeof(spdm_message_header_t)) {
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            break;
+        }
+
+        if (spdm_response->header.request_response_code == SPDM_ERROR
+            && spdm_response->header.param1 == SPDM_ERROR_CODE_LARGE_RESPONSE) {
+
+            /* It is possible that the CHUNK_SEND_ACK + chunk response is larger
+             * than the DATA_TRANSFER_SIZE. In this case an ERROR_LARGE_RESPONSE
+             * is returned directly in the response buffer rather than part of
+             * the CHUNK_SEND_ACK. Store this error response in scratch buffer
+             * to be handled when reading response */
+            if (response_size < send_info->large_message_capacity) {
+                libspdm_copy_mem(
+                    send_info->large_message, send_info->large_message_capacity,
+                    spdm_response, response_size);
+                send_info->large_message_size = response_size;
+            }
+            else {
+                status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+                break;
+            }
+        }
+        else {
+
+            if (spdm_response->header.spdm_version
+                != libspdm_get_connection_version(spdm_context)) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                break;
+            }
+            if (spdm_response->header.request_response_code != SPDM_CHUNK_SEND_ACK) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
+                break;
+            }
+
+            if (response_size < sizeof(spdm_chunk_send_ack_response_t)) {
+                status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+                break;
+            }
+            if (spdm_response->header.param1
+                & SPDM_CHUNK_SEND_ACK_RESPONSE_ATTRIBUTE_EARLY_ERROR_DETECTED) {
+                status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
```

Comment:
The error response will be present in `CHUNK_SEND_ACK.ResponseToLargeRequest`.

Reply:
Not clear what the ask is here?

Reply:
This means that the Responder has detected an error in the request and has returned the error to the Requester. libspdm needs to parse the error response and return that to the integrator: `LIBSPDM_STATUS_RESYNCH_PEER`, `LIBSPDM_STATUS_BUSY_PEER`, etc.

Reply:
Is the error response of type `spdm_error_response_t` which has an ERROR_CODE and ERROR_DATA? So this would be different type than `LIBSPDM_STATUS_RESYNC_PEER`, `LIBSPDM_STATUS_BUSY_PEER`, etc of type `libspdm_return_t`. I'm not clear on what mapping of values should be used between `spdm_error_response_t` and `libspdm_return_t`.

Reply:
I will save the `spdm_error_response_t` to the internal scratch space and make the this code break and return success. That will lead to the error response being read later on in the flow by `libspdm_receive_spdm_response`.